src/GraphCoarsen.cpp

```diff
@@ -298,12 +299,18 @@ void GraphCoarsen::BuildPEdges(
     NormalizeTraces(edge_traces, Agg_vertex, face_edge);
 
     std::unique_ptr<CoarseMBuilder> mbuilder_ptr;
```

Comment:
In order to reuse the components, the scope of `mbuilder_ptr` needs to be in `GraphCoarsen`, not just in this function? (this is probably in your TODO)

Reply:
Yes, that's what I mean by the third TODO item. I'm not sure exactly how to do it, but probably pass `mbuilder` to `BuildPEdges` instead of constructing it there. Right now I'm just demonstrating that this way of constructing coarse M works the same as the previous way.